### ISSUE TITLE
Add optional delays in collective pallet to increase security

### ISSUE THREAD

**Issue by gpestana** [paritytech/substrate#12622]
Adding block delays to when a new council can propose motions and execute approved motions can increase the security of the governance protocol by allowing the community (or sudo) to kick rogue councils and/or revert proposals. These delays should be optional.

- [ ] Add an optional delay between a council being elected and being able to propose motions;
- [ ] Add an optional delay between a council motion being approved and executed.

**Comment by bkchr** [comment]
Instead of having a delay, the council should not be able to execute anything that is related to moving funds or similar. Especially they should not get sudo rights.

**Comment by gpestana** [comment]
@bkchr that's a good point. Unfortunately that was exactly what happened in this incident https://blog.mangata.finance/blog/2022-10-08-council-incident-report/

The suggested delays would help to some extent but I agree that the root issue is the council having sido privileges.

**Comment by bkchr** [comment]
Yeah I knew what you were talking about, this is the reason I brought this up. For me this is a configuration error. Delays will not fix this. Council could also propose some change that is just a hash and send the preimage in the last moment. You wouldn't know what you want stop.

Just don't give council root origin. There are also million other ways to fuck up your runtime, you can not solve all of them.

**Comment by kianenigma** [comment]
A delay for council stuff makes less sense to me. 

A delay in the elections-phragmen pallet is not a bad idea. Essentially, you _announce_ the next council a few blocks before it is _enacted_.

**Comment by Polkadot-Forum** [comment]
This issue has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/thoughts-on-governance-security-and-configs-for-parachains/1166/1
